validations/scripts/factory/cluster_readiness_check.py

```diff
@@ -0,0 +1,180 @@
+        if node_res['ret_code']:
+            return node_res
+        nodes = node_res['response']
+        res = self.cluster.get_resource_failcount()
+        if res[1] == " ":
+            response["message"]= str(RESOURCE_FAIL_CHECK)
+            response["ret_code"]= 0
+        else:
+            response["message"]= str(RESOURCE_FAIL_ERROR)
+            response["ret_code"]= 1
+        response["response"]= res[1]
+        response["error_msg"]= res[2]
+        return response
+
+    def check_cluster_status(self):
+        ''' Validations for cluster status
+        '''
+        response = {}
+        node_res = PillarGet.get_pillar("cluster:node_list")
+        #get nodes
+        if node_res['ret_code']:
+            return node_res
+        nodes = node_res['response']
+        res = self.cluster.cluster_status()
+        print ("Cluster Response: " ,res)
+        if res[0]:
+            print ("TEST", response["message"])
+            response["message"]= str(CLUSTER_HEALTH_ERROR)
+        else:
+            for node in nodes:
+                if node in res[1]: 
+                    response["message"]= str(CLUSTER_HEALTH_CHECK)
+                    response["ret_code"]= 0
+                else:
+                    response["message"]= str(CLUSTER_HEALTH_ERROR)
+                    response["ret_code"]= 1
+        response["response"]= res[1]
+        response["error_msg"]= res[2]
+        return response
+
+    def check_stonith_issues(self):
+        ''' Validations for STONITH issues
+        '''
+        response = {}
+        node_res = PillarGet.get_pillar("cluster:node_list")
+        #get nodes
+        if node_res['ret_code']:
+            return node_res
+        res = self.cluster.stonith_issues()
+        print ("CHECK : ", res)
+        if (res[0][0] and res[1][0] and res[2][0]) == 0: 
+            response["message"]= str(STONITH_CHECK) 
+            response["ret_code"]= 0
+        else: 
+            response["message"]= str(STONITH_ERROR)
+            response["ret_code"]= 1
+        response["response"]= res[1][1]
+        response["error_msg"]= res[2][2]
+        return response
+
+    def ping_bmc(self):
+        ''' Ping BMC IP
+        '''
+        response = {}
+        res_ip = self.bmc.get_bmc_ip()
+        if not res_ip[0]:
+            ping_check = NetworkValidations.check_ping(res_ip)
+            if ping_check[0] == 0:
+                response["message"]= str(BMC_ACCESSIBLE_CHECK)
+            else:
+                response["message"]= str(BMC_ACCESSIBLE_ERROR)
+        else:
+            response["message"]= str(BMC_IP_ERROR)
+        response["ret_code"]= ping_check[0]
+        response["response"]= ping_check[1]
+        response["error_msg"]= ping_check[2]
+        return response
+
+    def check_bmc_accessible(self):
+        ''' Validations for BMC accessibility
+        '''
+        response = {}
+        bmc_res = self.bmc.bmc_accessible()
+        bmc_ping_res = self.bmc.ping_bmc() 
```

Comment:
Function bmc.bmc_accessible() already calls ping_bmc() along with checking power status. So no need to call ping_bmc() here again. Also check_bmc_accessible() function is already implemented in unboxing_readiness_check.py with same name. I would suggest to modify bmc.bmc_accessible() to bmc.check_power_status() to just check power status, so that you can call both (ping_bmc() and check_power_status()) functions separately whenever required. Ref verify_bmc_accessible() validations/scripts/factory/bmc_checks.py [PR #655 ](https://github.com/Seagate/cortx-prvsnr/pull/655/files)